be/src/olap/txn_manager.cpp

```diff
@@ -225,6 +235,7 @@ OLAPStatus TxnManager::commit_txn(
         WriteLock wrlock(&_txn_map_lock);
         TabletTxnInfo load_info(load_id, rowset_ptr);
         _txn_tablet_map[key][tablet_info] = load_info;
+        _insert_txn_partition_map(transaction_id, partition_id);
```

Comment:
redundant with prepare_txn()

Reply:
No, commit_txn() will be called when BE starts. So we need to rebuild _insert_txn_partition_map alongside with _txn_tablet_map.
And also, _txn_partition_map support to add same elements multi times, because it is a map with a set.